documentation/user_docs/docs/Advanced_Multifit.rst

```diff
@@ -0,0 +1,74 @@
+
+If multifit is being used to fit functions to objects rather than x-y-e triples, then there are some methods that need to be defined. You might want to fit the objects if their internal structure is more complex, for example if the fitting function depends on fields other than just the x values and parameters being passed to the fit function. Another case is when the masking of points from fitting requires manipulation of fields other than simply removing x-y-e values. [An example is the case of the sqw objects used in Horace. Here the calculation of the intensity at a data point depends on the information of the individual pixels that contribute to that data point. Masking requires that the pixel information of masked bins is removed from the sqw object.]
+
+The methods required for fitting objects with multifit are as follows:
+
+Fit functions
+*************
+
+The global function, and background function(s) if given, can be methods of the class or simply functions, with input argument form as described in detail in multifit help. The general format is:
+
+::
+
+        >> wcalc = my_function (w,p,c1,c2,...)
+
+If multifit is defined as a method of the class, then one can use the capability of nesting functions within the method to accept different fit function syntax. This is done, for example, for sqw objects when the fit functions to multifit, and equivalently multifit_func, are just a 1D/2D...4D Gaussian (according to the dimensionality of the sqw object).
+
+
+Utility methods
+---------------
+
+These are required to enable multifit to work with objects
+
+wout = mask (win, msk)
+~~~~~~~~~~~~~~~~~~~~~~
+
+A method that masks data points from further calculation. The output object must be a valid instance of the class in which the masked values have been removed in whatever sense the class requires.
+
+
+[y,var,msk] = sigvar_get(win)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A method that returns the intensity and variance arrays from the objects, along with a mask array that indicates which elements are to be retained (where elements of msk are true, the corresponding elements of y and var are retained). The output arrays y and var must have the same size and shape; msk must have the same number of elements (but can be a different shape). The array msk must be understood by the method 'mask' defined below.
+
+
+wsum = w1 + w2
+~~~~~~~~~~~~~~
+
+If a background function is provided, addition of objects must be defined as (requires overloading of the addition operator with a method named plus.m)
+
+
+
+**optional** x = sigvar_getx(win)
```

Comment:
It would be good to have sigvar_getx next to sigvar_get, and also cross-referenced in sigvar_get's description. It's not strictly optional; it is compulsory (yes?) if mask_points is not present.